src/OpenLoco/src/Ui/Widget.h

```diff
@@ -78,20 +86,59 @@ namespace OpenLoco::Ui
         static constexpr uint32_t kContentNull = 0xFFFFFFFFU;
         static constexpr uint32_t kContentUnk = 0xFFFFFFFEU;
 
-        WidgetType type;           // 0x00
-        WindowColour windowColour; // 0x01
-        int16_t left;              // 0x02
-        int16_t right;             // 0x04
-        int16_t top;               // 0x06
-        int16_t bottom;            // 0x08
+        constexpr Widget(Ui::Point32 origin, Ui::Size32 size, WidgetType widgetType, WindowColour colour, uint32_t content = Widget::kContentNull, StringId tooltip = StringIds::null)
+            : type{ widgetType }
+            , windowColour{ colour }
+            , left{ static_cast<int16_t>(origin.x) }
+            , right{ static_cast<int16_t>(origin.x + size.width - 1) }
+            , top{ static_cast<int16_t>(origin.y) }
+            , bottom{ static_cast<int16_t>(origin.y + size.height - 1) }
```

Comment:
Not necessary for this PR, but I think if we should work towards saving position and dimensions, instead of going for left/right/top/bottom. Is that your goal as well?

Reply:
I think so, not really a fan of using Rect.